cpp/lib/src/TimeUtil.cc

```diff
@@ -763,39 +763,271 @@ time_t UTCStructTmToTimeT(const struct tm &tm) {
+    timezone_abbreviation->insert(std::make_pair("VLAST", +11));
+    timezone_abbreviation->insert(std::make_pair("VLAT", +10));
+    timezone_abbreviation->insert(std::make_pair("VOST", +6));
+    timezone_abbreviation->insert(std::make_pair("VUT", +11));
+    timezone_abbreviation->insert(std::make_pair("W", -10));
+    timezone_abbreviation->insert(std::make_pair("WAKT", +12));
+    timezone_abbreviation->insert(std::make_pair("WARST", -3));
+    timezone_abbreviation->insert(std::make_pair("WAST", +2));
+    timezone_abbreviation->insert(std::make_pair("WAT", +1));
+    timezone_abbreviation->insert(std::make_pair("WEST", +1));
+    timezone_abbreviation->insert(std::make_pair("WET", +0));
+    timezone_abbreviation->insert(std::make_pair("WFT", +12));
+    timezone_abbreviation->insert(std::make_pair("WGST", -2));
+    timezone_abbreviation->insert(std::make_pair("WGT", -3));
+    timezone_abbreviation->insert(std::make_pair("WIB", +7));
+    timezone_abbreviation->insert(std::make_pair("WIT", +9));
+    timezone_abbreviation->insert(std::make_pair("WITA", +8));
+    timezone_abbreviation->insert(std::make_pair("WST", +13));
+    timezone_abbreviation->insert(std::make_pair("WST", +1));
+    timezone_abbreviation->insert(std::make_pair("WT", +0));
+    timezone_abbreviation->insert(std::make_pair("X", -11));
+    timezone_abbreviation->insert(std::make_pair("Y", -12));
+    timezone_abbreviation->insert(std::make_pair("YAKST", +10));
+    timezone_abbreviation->insert(std::make_pair("YAKT", +9));
+    timezone_abbreviation->insert(std::make_pair("YAPT", +10));
+    timezone_abbreviation->insert(std::make_pair("YEKST", +6));
+    timezone_abbreviation->insert(std::make_pair("YEKT", +5));
+    timezone_abbreviation->insert(std::make_pair("Z", +0));
+}
 
 // See https://www.rfc-editor.org/rfc/rfc822.txt section 5.1.
 static bool ZoneAdjustment(const std::string &rfc822_zone, time_t * const adjustment) {
-    if (rfc822_zone == "GMT" or rfc822_zone == "UT")
-        *adjustment = 0;
-    else if (rfc822_zone == "EST")
-        *adjustment = +5 * 3600;
-    else if (rfc822_zone == "EDT")
-        *adjustment = +4 * 3600;
-    else if (rfc822_zone == "CST")
-        *adjustment = +6 * 3600;
-    else if (rfc822_zone == "CDT")
-        *adjustment = +5 * 3600;
-    else if (rfc822_zone == "MST")
-        *adjustment = +7 * 3600;
-    else if (rfc822_zone == "MDT")
-        *adjustment = +6 * 3600;
-    else if (rfc822_zone == "PST")
-        *adjustment = +8 * 3600;
-    else if (rfc822_zone == "PDT")
-        *adjustment = +7 * 3600;
-    else if (rfc822_zone == "A")
-        *adjustment = -1 * 3600;
-    else if (rfc822_zone == "M")
-        *adjustment = -12 * 3600;
-    else if (rfc822_zone == "N")
-        *adjustment = +1 * 3600;
-    else if (rfc822_zone == "Y")
-        *adjustment = +12 * 3600;
-    else // Unrecognized tiem zone.
-        return false;
+    std::map<std::string, float> tmz_map;
```

Comment:
This should be static and the Construct... should be called with constructor notation